streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -62,9 +60,9 @@ public static StoreFactory create(final String name) {
                                     @Override
                                     public StateStoreSupplier build() {
                                         if (capacity < Integer.MAX_VALUE) {
-                                            return new InMemoryLRUCacheStoreSupplier<>(name, capacity, serdes, null);
+                                            return new InMemoryLRUCacheStoreSupplier<>(name, capacity, keySerde, valueSerde, null);
                                         }
-                                        return new InMemoryKeyValueStoreSupplier<>(name, serdes, null);
+                                        return new InMemoryKeyValueStoreSupplier<>(name, keySerde, valueSerde, null);
```

Comment:
Also, I realized that we apparently have the `Time time` field only because we need it to construct a `MeteredKeyValueStore` instance, where `time == null` will result in the use of `SystemTime()`:

```java
   // always wrap the store with the metered store
    public MeteredKeyValueStore(final KeyValueStore<K, V> inner, String metricScope, Time time) {
        this.inner = inner;
        this.metricScope = metricScope;
        this.time = time != null ? time : new SystemTime();
    }
```

So my comments above are slightly wrong, though I think they still capture what I wanted to say:  we should come up with a better and semantically clearer way in the code than passing magic `null`'s around.